Update RHEL and Clones compilation instructions

juliogonzalez opened the pull request:
The [CI started failing for Rocky Linux 8 on August 9th](https://jenkins.juliogonzalez.es/job/tds_fdw-build/3236/console) as `/usr/bin/clang` was not installed. I tried installing from scratch and more packages were missing besides `clang` (`llvm` and `redhat-rpm-config`, this one at least on containers).

Since `gcc` is a requirement for `clang`,  we don't need to specify it.

So this PR is basically implementing at the doc, the following changes at the images: https://github.com/tds-fdw/ci-setup/compare/b3d2ac6ec21bf899c6ea56d6b6aed47e8727cb7a...399c0df2073872242404818b2dcdae3c7538ab65

With that, the extension builds and the test are able to run: https://jenkins.juliogonzalez.es/view/tds_fdw-jobs/job/tds_fdw-build/3243/

I am not sure why this is suddenly happening, I guess it's a change on the dependency tree, maybe on PostgreSQL itself not requiring `clang` and `lvm` for the devel package.

Besides this, I removed the instructions about how to install PostgreSQL, as the official doc that is linked does a perfect job, with a form to pick architecture, platform, etc, and this way we don't need to worry about using the right repositories or GPG keys.

juliogonzalez commented on InstallRHELandClones.md:
````diff
@@ -79,7 +55,13 @@ sudo yum install epel-release
+sudo yum install clang llvm make redhat-rpm-config wget
+```
+
+#### IMPORTANT: CentOS7/Oracle7 and PostgreSQL >= 11
````
Let's not remove CentOS7 for now. It's EoL, but there are still some alive clones there, and the PostgreSQL development group still list it at their installation instructions.

juliogonzalez commented on InstallRHELandClones.md:
````diff
@@ -53,33 +39,29 @@ sudo yum install tds_fdw11.x86_64
-sudo rpm -i https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-x86_64/pgdg-redhat-repo-latest.noarch.rpm
-sudo yum install postgresql11 postgresql11-server postgresql11-libs postgresql11-devel
-```
+You need to enable the PowerTools repository for RHEL8 and clones, or the CBR repository for RHEL9 and clones.
````
Required as `postgresqlXX-devel` requires at least a perl dependency from there.

juliogonzalez commented:
Ready to be reviewed.

GeoffMontee commented:
Thanks for the fixes!

devrimgunduz commented:
Why did you remove RPM repo installation instructions?  We fully support tds_fdw in the repo.

GeoffMontee commented:
Hi @devrimgunduz,

This PR removed the instructions that showed how to configure the repo and how to install PostgreSQL from the repo. This is because the page links to [this page](https://www.postgresql.org/download/linux/redhat/), which already provides great instructions on how to configure the repo and how to install PostgreSQL from it.

Regardless, the instructions to install tds_fdw from the repo are still present here:

https://github.com/tds-fdw/tds_fdw/blob/master/InstallRHELandClones.md#tds_fdw

If you think this should be improved, please let us know. Thanks!

juliogonzalez commented:
@devrimgunduz, as @GeoffMontee explained, the instructions are not removed, we still link the official doc at https://github.com/tds-fdw/tds_fdw/pull/366/files#diff-4347d35cd8bd8504b6a9a8e7d3eaab6131fe5858c899c97a05268c508a06ce95R18

What we removed was the specific example for EL8 and PostgreSQL.

If the user wants to install PostgreSQL, the link at https://github.com/tds-fdw/tds_fdw/pull/366/files#diff-4347d35cd8bd8504b6a9a8e7d3eaab6131fe5858c899c97a05268c508a06ce95R18 does a perfect job.

And then https://github.com/tds-fdw/tds_fdw/pull/366/files#diff-4347d35cd8bd8504b6a9a8e7d3eaab6131fe5858c899c97a05268c508a06ce95R22 still points to the package the PostgreSQL development group provides.

What I fixed are the instructions to compile, as they didn't work anymore (I detected it with the CI)

As Geoff said, feel free to improve the doc, it's just a PR away :-)
